src/implementation/data-elements-and-custom-dimensions.md

```diff
@@ -122,14 +146,81 @@ renders these combinations invalid and a low-level database operation
+
+Thus, while attribute option combinations are structually equivalent to category option combinations, they are used to disaggregate data at the level of the data set. All data value which are part of a data set which is associated with an attribute combination, would be recorded and disaggregated with an additional fifth dimension, in addition to the four primary dimensions listed above. There are no restrictions on how an attribute combination can be constructed, which in turn allows for implementers to design arbitrary dimensions for specific data sets. 
+
+
+## Group sets and analytical dimensions
+
+Category and attribute combinations are used during data entry to disaggretgate data in certain ways, 
+for instance by age and sex breakdowns. When the data is later analyzed, there may be a need to 
+aggregate or group the data in different ways. Consider a category with the following age bands: 
+
+* \< 1
+* 1-4
+* 5-10
+* 10-15
+* 15-19
+* 20-29
+* 30-49
+* 49+
+
+Data may be entered with these age groups, but when analyzed in the analytics apps of DHIS2, 
+there may be a need to group the data according to more coarse age bands.
+Using a category option group set, we could create two category groups, such as \<15 and 15+. 
+Each of the original category options could then be placed into the corresponding
+category option group. Each of the groups can then be associated with a category option group 
+set, which becomes available as an additional dimension in the analytics apps. 
+
+
+Category option group sets are particularly useful for creating higher level groupings of common
+category options. This approach is often useful for combining data elements which may have been 
+collected according to related, but different category combinations. 
+
+
 ## Data element groups
 
-Common properties of data elements can be modelled through what is
-called data element groups. The groups are completely flexible in the
-sense that both their names and their memberships are defined by the
-user. Groups are useful both for browsing and presenting related data,
+Data elements which are related to one another can be grouped together
+with a *data element group*. Data element groups can. Data element 
+groups are completely flexible in the  sense that both their names and their
+ memberships are completely flexible in how they are designed. 
```

Comment:
Unfinished sentence "Data element groups can.", plus the "completely flexible" sentence reads awkwardly and doesn't really improve on the previous version.

Reply:
```suggestion
with a *data element group*. Data element groups are completely flexible in
terms of both their names and their memberships. 
```